Inline inputs for checkbox and radio buttons

delisma opened the pull request:
CSS changes and coded documentation for inline checkboxes and radio buttons inputs in `<div>` elements and `<li>` elements

duboisp commented on src/plugins/formvalid/_base.scss:
```diff
@@ -46,5 +46,31 @@
+		}
 	}
 }
+ul.form-group {
```
This style do not belong to the formvalid plugin, instead it should be more global, like the bootstrap override

GormFrank replied:
Put those styles inside another file, do not encapsulate them inside formvalid

duboisp commented on Gruntfile.coffee:
```diff
@@ -1009,6 +1009,7 @@ module.exports = (grunt) ->
 						"W010" # Using `.pull-left` or `.pull-right` as part of the media object component is deprecated as of Bootstrap v3.3.0. Use `.media-left` or `.media-right` instead.
 						"E013" # Only columns (`.col-*-*`) may be children of `.row`s
 						"E014" # Columns (`.col-*-*`) can only be children of `.row`s or `.form-group`s
+						"E017" # Support for checkboxes with explicit labels instead of implicit labels
```
We will move this exception for the formvalid page only.

EricDunsworth replied:
In case it helps: What @duboisp is suggesting would be very similar to what I recently did in wet-boew/GCWeb#1605

GormFrank commented on package.json:
```diff
@@ -34,6 +34,7 @@
   "license": "MIT",
   "dependencies": {
     "bootstrap-sass": "3.4.1",
+    "bower": "^1.8.8",
```
Please do not include package.json and package-lock.json to your PR.

GormFrank commented on Gruntfile.coffee:
```diff
@@ -1009,6 +1009,7 @@ module.exports = (grunt) ->
 						"W010" # Using `.pull-left` or `.pull-right` as part of the media object component is deprecated as of Bootstrap v3.3.0. Use `.media-left` or `.media-right` instead.
 						"E013" # Only columns (`.col-*-*`) may be children of `.row`s
 						"E014" # Columns (`.col-*-*`) can only be children of `.row`s or `.form-group`s
+						"E017" # Support for checkboxes with explicit labels instead of implicit labels
```
We'll have to look into a new rule for this use case. (checkbox AND radio included)

duboisp reviewed (changes requested):
The update to package.json and package-lock.json would need to be removed from this PR.

duboisp reviewed (commented):
Todo:
* Test accessibility - Check if the design pattern that use unordered list of "radio" or "checkbox" is acceptable
* Test accessibility - Check if the design pattern that use ```<ul>``` inside a ```fieldset``` is too much
* Test accessibility - Check if we only use ```div``` for a list of ```input[type=radio]``` and  ```input[type=radio]```
* Test accessibility - Is radio and checkbox must be always in a fieldset?
* Test accessibility - When radio and checkbox are not in a fieldset, can they be put in a list

Bootlint don't seems to support explicit labels. Explicit labels is needed to enable a improved visual for bigger and flat checkbox/radio that would improve UX.

duboisp commented:
@bsouster as discussed, you will find the pattern in the last example in this demo
https://universallabs.org/wet/labs/pr8744/unmin/demos/formvalid/formvalid-en.html

Also, see my [previous comments](https://github.com/wet-boew/wet-boew/pull/8744#pullrequestreview-305986836) regarding what we need to test and validate

EricDunsworth commented:
FWIW this is how Bootstrap 3.4's documentation recommends doing layouts for horizontal checkboxes/radio buttons:
https://getbootstrap.com/docs/3.4/css/#inline-checkboxes-and-radios

GormFrank commented on src/base/bootstrap-overrides/_base.scss:
```diff
@@ -589,6 +589,24 @@ input[type=button], input[type=reset], input[type=submit] {
+ */
+.form-inline {
+	.label-inline {
+		font-weight: 400;
```
There might be a variable available for this one.

delisma replied:
I've checked all the scss files to find a variable that set a font-weight to normal but nothing seems to fit the needs.

GormFrank commented on src/base/bootstrap-overrides/_base.scss:
```diff
@@ -589,6 +589,24 @@ input[type=button], input[type=reset], input[type=submit] {
+		vertical-align: middle;
+	}
+	.form-group + .form-group {
+		padding-left: 10px;
```
The mrgn-lft-X class should be used instead.

delisma replied:
The class mrgn-lft-x adds margin not padding.

GormFrank commented on src/base/bootstrap-overrides/_base.scss:
```diff
@@ -648,6 +666,9 @@ fieldset {
 			margin-bottom: 5px;
 		}
+		input + label {
+			padding: 0 1ch;
```
Update this one

delisma replied:
Done

GormFrank reviewed (changes requested):
Also don't forget to squash your commits.

GormFrank commented:
Pre-approved during technical review conditionally to testing and @duboisp accepting changes.

duboisp commented on src/base/bootstrap-overrides/_base.scss:
```diff
@@ -589,6 +589,35 @@ input[type=button], input[type=reset], input[type=submit] {
+		}
+	}
+}
+@media (min-width: $screen-sm-min) {
```
Move this in the view or create one for bootstrap and add it the main SCSS include file

Commit:
Adjusted implicit and explicit labels patterns

Improved explicit labels support for checkbox and radio button inputs


Improved explicit labels support for checkbox and radio button inputs

Improved explicit labels support for checkbox and radio button inputs


Improved explicit labels support for checkbox and radio button inputs

 Improved explicit labels support for checkbox and radio button inputs


Indent issues

Adjusted implicit and explicit labels patterns

Improved explicit labels support for checkbox and radio button inputs


Improved explicit labels support for checkbox and radio button inputs

Improved explicit labels support for checkbox and radio button inputs


Improved explicit labels support for checkbox and radio button inputs

 Improved explicit labels support for checkbox and radio button inputs


Indent issues

Correction for tavis-ci

Added mobile view support

Adjusted implicit and explicit labels patterns


Adjusted implicit and explicit labels patterns


Adjusted implicit and explicit labels patterns


Adjusted implicit and explicit labels patterns

duboisp commented:
Pre-approved waiting for Travis-CI biuld to pass.
